Assignment4/NumberSet/NumSet.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NumberSet
+{
+    public class NumSet
+    {
+        private HashSet<int>? numberSet;
+
+
+        public NumSet(params int[] ? numSet)
+        {
+            if(numSet is null)
+            {
+                throw new ArgumentNullException(nameof(NumSet));
+            }
+            numberSet = new HashSet<int>(numSet);
+        }
+
+        public override string ToString()
+        {
+
+            if(numberSet is null)
+            {
+                throw new ArgumentNullException(nameof(NumSet));
+            }
+
+            return String.Join(",", numberSet);

```

Comment:
```suggestion
            return string.Join(",", numberSet);
```
Use C# keywords rather than BCL names.